### PR TITLE
Rcpp implementation of final_size

### PR DESCRIPTION
**Pull request by pratikunterwegs**
An implementation of the current epidemic final size code in `finalsize.R` in C++ using RcppEigen. Adds the function `final_size_cpp`, with the same arguments as `final_size`.

Fixes:
- #8 by implementing a very basic solver algorithm
- #34 by adding `catch2` based testing for C++ funs in header `finalsize.h`. Catch2 used for better integration with `testthat`
- #35 by adding regular `testthat` tests

### CODE REVIEW

File: src/finalsize.cpp [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+                               const Eigen::MatrixXd &contact_matrix,
+                               const Eigen::VectorXd &demography_vector,
+                               const Eigen::VectorXd &prop_initial_infected,
+                               Eigen::VectorXd prop_suscep,
+                               const int iterations = 30) {
+  if (prop_suscep.size() == 1) {
+    const double prop_suscep_ = prop_suscep[0];
+    prop_suscep.resize(demography_vector.size(), 1);
+    prop_suscep.fill(prop_suscep_);
+    Rcpp::Rcout << "prop_suscep = " << prop_suscep << "\n";
```
Comment (Bisaloo) [inline src/finalsize.cpp:46]: Do we want a message here? We didn't have it in the R implementation

**Review by Bisaloo — APPROVED** [review]
LGTM! Just left 2 minor comments

File: R/finalsize-package.R [hunk 2]
```diff
@@ -0,0 +1,5 @@
+#' @useDynLib finalsize
+#' @import RcppEigen
+#' @importFrom Rcpp evalCpp
+#' @exportPattern "^[[:alpha:]]+"
```
Comment (Bisaloo) [inline R/finalsize-package.R:4]: Can we remove this as well? I think it's safer to export select objects.

File: DESCRIPTION [hunk 3]
```diff
@@ -38,10 +38,17 @@ Encoding: UTF-8
 Suggests: 
     covr,
     socialmixr,
-    testthat (>= 3.0.0)
+    testthat (>= 3.0.0),
+    xml2
```
[the remark this answers is not in the document]
Comment (pratikunterwegs) [reply DESCRIPTION:42]: Recommended addition when using `testthat::use_catch` to set up `catch2` unit testing

File: src/finalsize.cpp (hunk 1 above)
Comment (pratikunterwegs) [reply src/finalsize.cpp]: Removed in https://github.com/epiverse-trace/finalsize/pull/36/commits/e12cf683ddba150a3d37a87581219ce38bc7cc76

File: R/finalsize-package.R (hunk 2 above)
Comment (pratikunterwegs) [reply R/finalsize-package.R]: Removed in https://github.com/epiverse-trace/finalsize/pull/36/commits/940fb38c0814679a04b9d5f9af2c393378952f23

**Pull request merged**